sdk/cosmos/azure-cosmos-spark_3-3_2-12/CHANGELOG.md

```diff
@@ -10,287 +11,4 @@
-
-#### Bugs Fixed
-* Added validation for all config-settings with a name starting with "spark.cosmos."
-* Fixed a bug in bulk write causing nonresponse.
-
-### 4.0.0-beta.2 (2021-04-19)
-* Cosmos DB Spark 3.1.1 Connector Preview `4.0.0-beta.2` Release.
-
-#### New Features
-* The beta-2 is feature-complete now
-* Spark structured streaming (micro batches) for consuming change feed
-* Spark structured streaming (micro batches) support added for writes (TableCapability.STREAMING_WRITE)
-* Allowing configuration of "Cosmos views" in the Spark catalog to enable direct queries against Spark catalog
-
-#### Bugs Fixed
-* Perf validation and optimizations (resulting in significant better throughput for read code path)
-* Row conversion: Allow configuration of behavior on schema mismatch - error vs. null
-* Row conversion: Supporting InternalRow type to avoid failures when using nested StructType of InternalRow (not Row)
-
-#### Known limitations
-* No support for continuous processing (change feed) yet. (will be added after GA)
-* No perf tests / optimizations have been done yet - we will iterate on perf in the next preview releases. So usage should be limited to non-production environments with this preview.
-
-### 4.0.0-beta.1 (2021-03-22)
-* Cosmos DB Spark 3.1.1 Connector Preview `4.0.0-beta.1` Release.
-#### Features
-* Supports Spark 3.1.1 and Scala 2.12.
-* Integrated against Spark3 DataSourceV2 API.
-* Devloped ground up using Cosmos DB Java V4 SDK.
-* Added support for Spark Query, Write, and Streaming.
-* Added support for Spark3 Catalog metadata APIs.
-* Added support for Java V4 Throughput Control.
-* Added support for different partitioning strategies.
-* Integrated against Cosmos DB TCP protocol.
-* Added support for Databricks automated Maven Resolver.
-* Added support for broadcasting CosmosClient caches to reduce bootstrapping RU throttling.
-* Added support for unified jackson ObjectNode to SparkRow Converter.
-* Added support for Raw Json format.
-* Added support for Config Validation.
-* Added support for Spark application configuration consolidation.
-* Integrated against Cosmos DB FeedRange API to support Partition Split Proofing.
-* Automated CI testing on DataBricks and Cosmos DB live endpoint.
-* Automated CI Testing on Cosmos DB Emulator.
-
-#### Known limitations
-* Spark structured streaming (micro batches) for consuming change feed has been implemented but not tested end-to-end fully so is considered experimental at this point.
-* No support for continuous processing (change feed) yet.
-* No perf tests / optimizations have been done yet - we will iterate on perf in the next preview releases. So usage should be limited to non-production environments with this preview.
-
-### 4.0.0-alpha.1 (2021-03-17)
-* Cosmos DB Spark 3.1.1 Connector Test Release.
+### NOTE: See CHANGELOG.md in 3.1 and 3.2 projects for changes prior to 3.3
```

Comment:
Do you mean prior to 4.15.0 ?

Reply:
Change log will be maintained for each Spark release version - so if we fix something it will show up in changelog for Spark 3.1, 3.2 and 3.3 - if 3.3 is introduced in 4.15 it is ok to refer to external files for < 4.15 - but starting 4.15 all changes need to be triplicated

Reply:
Makes sense, will change to refer to 4.15.0

Reply:
Fixed